packages/lexical-link/src/__tests__/unit/LexicalLinkNode.test.ts

```diff
@@ -409,5 +412,70 @@ describe('LexicalLinkNode tests', () => {
       const link = paragraph.children[0] as SerializedLinkNode;
       expect(link.title).toBe('Lexical Website');
     });
+
+    test('$toggleLink correctly removes link when textnode has children(like marknode)', async () => {
+      const {editor} = testEnv;
+      await editor.update(() => {
+        const paragraph = $createParagraphNode();
+        const precedingText = $createTextNode('some '); // space after
+        const textNode = $createTextNode('text');
+
+        paragraph.append(precedingText, textNode);
+
+        const linkNode = $createLinkNode('https://example.com/foo', {
+          rel: 'noreferrer',
+        });
+        textNode.insertAfter(linkNode);
+        linkNode.append(textNode);
+
+        const markNode = $createMarkNode(['knetk']);
+        textNode.insertBefore(markNode);
+        markNode.append(textNode);
+        $getRoot().append(paragraph);
+      });
+
+      await editor.read(() => {
```

Comment:
No need to await for reads, nothing asynchronous is happening here

```suggestion
      editor.read(() => {
```